.github/workflows/pr_validate.yaml

```diff
@@ -0,0 +1,172 @@
+
+          SANDBOX=$(pcregrep -o2 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+          )
+          echo "Extracted sandbox:"
+          echo $SANDBOX
+          echo "::set-output name=sandbox::$SANDBOX"
+
+      - id: reach-staging
+        name: Attempt to reach staging environments
+        uses: nick-invision/retry@v2
+        with:
+          timeout_seconds: 30
+          max_attempts: 5
+          retry_on: error
+          command: curl -f ${{ steps.extract-staging.outputs.url }}
+          # Add jitter to break up correlated events.
+          on_retry_command: sleep $[($RANDOM % 10) + 5]s
+
+      - id: check-e2e-trigger
+        continue-on-error: true
+        name: Check if E2E testing is triggered
+        run: |-
+          REGEX="\[[Xx]\] Started end-to-end tests on the latest commit"
+    
+          pcregrep -o1 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+
+  run-integration-tests:
+    name: Run integration tests
+    runs-on: ubuntu-20.04
+    needs: lint-pr-body
+    if: needs.lint-pr-body.outputs.should_e2e_run == 'success'
+    steps:
+      - id: install
+        name: Install requirements
+        run: |-
+          sudo apt-get update -y
+          sudo apt-get install -y pcregrep
+
+      - id: extract-commit-details
+        name: Get details of last commit
+        run: |-
+          echo "SHA of GitHub merge commit:"
+          echo $GITHUB_SHA
+          echo "::set-output name=github_sha::$GITHUB_SHA"
+
+          REPO_NAME=$(echo $GITHUB_REPOSITORY | awk -F '/' '{print $2}')
+          echo "Repo name:"
+          echo $REPO_NAME
+          echo "::set-output name=repo_name::$REPO_NAME"
+
+      - id: extract-integration-test-ref
+        name: Get integration test ref
+        run: |-
+          REGEX="#### Integration test branch\s?\r\n(.+)\s?\r\n"
+
+          echo "REGEX to test against:"
+          echo $REGEX
+          echo "Now looking at regex result:"
+
+          INTEGRATION_TEST_REF=$(pcregrep -o1 -M "$REGEX" <<\EOF
+          ${{ github.event.pull_request.body }}
+          EOF
+          )
+
+          echo "Ref given is $INTEGRATION_TEST_REF, setting it as is."
+          echo "::set-output name=ref::$INTEGRATION_TEST_REF"
+
+      # - id: run-integration-test
+      #   name: Run integration tests
+      #   uses: aurelien-baudet/workflow-dispatch@v2
+      #   with:
+      #     workflow: Run end-to-end tests
+      #     repo: biomage-ltd/testing
+      #     token: ${{ secrets.API_TOKEN_GITHUB }}
+      #     ref: ${{ steps.extract-integration-test-ref.outputs.ref }}
+      #     inputs: '{ "environment": "staging", "ref": "${{ steps.extract-integration-test-ref.outputs.ref }}", "sandboxId": "${{needs.lint-pr-body.outputs.sandbox}}", "image_sha" : "${{ steps.extract-commit-details.outputs.github_sha }}", "namespace": "${{steps.extract-commit-details.outputs.repo_name}}-${{needs.lint-pr-body.outputs.sandbox}}" }'
```

Comment:
did you comment this out for debugging purposes?

Reply:
yes